### PR TITLE
fix:not delete schedule completely

### PR DESCRIPTION
**Pull request by worldmaomao**
Fix the issue that : not delete the keys of scheduler action completely in redis. 
Avoid the memory leak

### CODE REVIEW

File: internal/pkg/db/redis/queries.go [hunk 1]
```diff
@@ -242,7 +242,9 @@ func deleteObject(remover models.Remover, id string, conn redis.Conn) {
 	for _, cmd := range remover.Remove() {
 		switch cmd.Command {
 		case "ZREM":
+			_ = conn.Send(cmd.Command, cmd.Hash, cmd.Key)
 		case "SREM":
+			_ = conn.Send(cmd.Command, cmd.Hash, cmd.Key)
 		case "HDEL":
 			_ = conn.Send(cmd.Command, cmd.Hash, cmd.Key)
```
Comment (michaelestrin) [inline internal/pkg/db/redis/queries.go]: Why duplicate these lines?  Why not `fallthrough`?

File: internal/pkg/db/redis/models/interval_action.go [hunk 2]
```diff
@@ -38,7 +38,6 @@ func NewIntervalAction(from contract.IntervalAction) (ia IntervalAction) {
 }
 
 func (ia IntervalAction) Add() (cmds []DbCommand) {
-	cmds = make([]DbCommand, len(intervalActionKeys))
```
Comment (michaelestrin) [inline internal/pkg/db/redis/models/interval_action.go]: Why did you remove the slice initialization (it's explicitly initializing the slice for the proper size)?

File: internal/pkg/db/redis/models/interval_action.go [hunk 3]
```diff
@@ -51,12 +50,11 @@ func (ia IntervalAction) Add() (cmds []DbCommand) {
 			cmds = append(cmds, DbCommand{Command: "SADD", Hash: key + ":" + ia.Target, Key: ia.ID})
 		}
 	}
-	return cmds
+	return
 }
 
 func (ia IntervalAction) Remove() (cmds []DbCommand) {
-	cmds = make([]DbCommand, len(intervalActionKeys))
```
Comment (michaelestrin) [inline internal/pkg/db/redis/models/interval_action.go]: Why did you remove the slice initialization (it's explicitly initializing the slice for the proper size)?

File: internal/pkg/db/redis/queries.go (hunk 1 above)
Comment (worldmaomao) [reply internal/pkg/db/redis/queries.go]: The original code is  only available for  redis command "HDEL".  
Command "SREM" and "SREM" is not available and will not send to redis.

File: internal/pkg/db/redis/models/interval_action.go (hunk 3 above)
Comment (worldmaomao) [reply internal/pkg/db/redis/models/interval_action.go]: If length of intervalActionKeys is 3, after slice initialization  there are 3 empty DbCommand structs. According to the below code, it will append more DbCommand struct  which will not replace the empty struct in the slice.
111WX20191231-134922@2x

File: internal/pkg/db/redis/models/interval_action.go (hunk 2 above)
Comment (worldmaomao) [reply internal/pkg/db/redis/models/interval_action.go]: If length of intervalActionKeys is 3, after slice initialization  there are 3 empty DbCommand structs. According to the below code, it will append more DbCommand struct  which will not replace the empty struct in the slice.
111WX20191231-134922@2x

File: internal/pkg/db/redis/queries.go (hunk 1 above)
Comment (worldmaomao) [reply internal/pkg/db/redis/queries.go]: That's a good idea.   
I will fix it.

File: internal/pkg/db/redis/models/interval_action.go (hunk 2 above)
Comment (michaelestrin) [reply internal/pkg/db/redis/models/interval_action.go]: You are correct.  However, I wonder if a more efficient solution would be to initialize cmds using make once (since we know its required size in advance) and subsequently assign its individual elements directly (as opposed to using append).

File: internal/pkg/db/redis/models/interval_action.go (hunk 3 above)
Comment (michaelestrin) [reply internal/pkg/db/redis/models/interval_action.go]: You are correct.  However, I wonder if a more efficient solution would be to initialize cmds using make once (since we know its required size in advance) and subsequently assign its individual elements directly (as opposed to using append).
